packages/botfuel-dialog/src/adapters/shell-adapter.js

```diff
@@ -57,6 +59,18 @@ class ShellAdapter extends Adapter {
     // eslint-disable-next-line no-console
     console.log(chalk.hex('#16a085')(`${DELIMITER}${botMessage.payload.value}`));
   }
+
+  /** @inheritDoc */
+  addProperties(message) {
+    return extend(
+      {
+        id: uuidv4(),
```

Comment:
it would be nice to have  extendMessageWithUUID and extendMessageWithTimestamp methods in the adapter class, you could then write for example:

extendMessage(message) {
   let extendedMessage = this.extendMessageWithUUID(message);
   extendedMessage = this.extendMessageWithTimestamp(extendedMessage);
   return { ...extendedMessage, adapter='shell' };
}

Hence, the uuid and timestamp generation mechanisms are in a single place
Or even better you could have this logic in a super method (but don't forget to override it in the test adapter)

Reply:
What do you think about methods `getMessageUUID()` and `getMessageTimestamp()` that just returns an uuid and a timestamp ?

Then use them like:

```javascript
extendMessage(message) {
  return {
    id: this.getMessageUUID(),
    timestamp: this.getMessageTimestamp(),
    adapter: 'adapter',
  };
}
```

Reply:
By the way I'm asking myself about the **id** and the **timestamp**, if we want to use them with botmeter maybe we should extend them directly in the adapter and not let the possibility for a bot developer to not add them in his own adapter ?